wesgi/__init__.py

```diff
@@ -144,27 +146,32 @@ def delete(key):
 
 class MiddleWare(object):
 
-    def __init__(self, app, policy='default', debug=True):
+    def __init__(self, app, policy='default', forward_headers=False, debug=True):
         self.debug = debug
         self.app = app
         if isinstance(policy, basestring):
             policy = _POLICIES[policy]
         self.policy = policy
+        self.policy.forward_headers = forward_headers
         self.http = policy.http()
 
     def __call__(self, environ, start_response):
         req = webob.Request(environ)
+        headers = {}
+        if self.policy.forward_headers:
+            headers = dict(req.headers.items())
+
         resp = req.get_response(self.app)
         if resp.content_type == 'text/html' and resp.status_int == 200:
             orig_scheme = environ['wsgi.url_scheme']
-            new_body = self._process(resp.body, orig_scheme)
+            new_body = self._process(resp.body, orig_scheme, headers=headers)
             if new_body is not None:
                 resp.body = new_body
         return resp(environ, start_response)
 
-    def _process(self, body, orig_scheme):
+    def _process(self, body, orig_scheme, headers={}):
```

Comment:
please don't use {} as the default of a keyword argument. That's a bug magnet as the dict is only instantiated once.

Please do something like this:
```
def _process(self, body, orig_scheme, headers=None):
    if headers is None:
        headers = {}
```
I'm probably missing something, but I don't see why you need to use a keyword argument over a positional argument here.